lib/Varien/Image/Adapter/Gd2.php

```diff
@@ -54,7 +54,9 @@ public function __construct()
      */
     public function destruct()
     {
-        @imagedestroy($this->_imageHandler);
+        if (is_resource($this->_imageHandler)) {
```

Comment:
Ref https://php.watch/versions/8.0/gdimage for PHP 8, not sure if it is applicable

```suggestion
        if (is_resource($this->_imageHandler) || $this->_imageHandler instanceof \GdImage) {
```